index.tt

```diff
@@ -1,157 +1,254 @@
+          <li>clean up everything that is not needed at runtime</li>
+          <li>decide how to split into layers for better caching</li>
+        </ul>
+        <p>
+          Writing <code>Dockefile</code> that would produce a minimal image is
+          at best a very error prone process.
+        </p>
+        <p>
+          With Nix you only include to docker image packages that you define.
+          No cleaning up needed. There are no build tools left in your docker
+          image, making it as minimal as you need.
+        </p>
+        <p>
+          Nix also know how to layer your resulting docker image, automatically
+          without you needed to do anything. The resulting layers are optimized
+          for caching as much as possible.
+        </p>
+        <p>
+          The following Nix expression (<code>default.nix</code>) defines a
+          docker image with <strong>only</strong> <code>hello</code> package
+          in it.
+        </p>
+        <pre class="well">
+{ pkgs ? import &lt;nixpkgs&gt; {}
+}:
+pkgs.dockerTools.buildLayeredImage {
+  name = "only-hello";
+  contents = [ pkgs.hello ];
+}</pre>
+        <p>To build and run the image you need to:</p>
+        <pre class="well">
+<span class="nix-command"><strong>nix-build default.nix -o ./result</strong></span>
+...
+/nix/store/&#8230;-docker-image-only-hello.tar.gz
+<span class="nix-command">docker load -i ./result</span>
+1c31fbac2eb1: Loading layer [==================>]  1.649MB/1.649MB
+03b22f688054: Loading layer [==================>]    256kB/256kB
+29c350a9c392: Loading layer [==================>]  31.61MB/31.61MB
+6a87e4d71e07: Loading layer [==================>]  266.2kB/266.2kB
+c09c43a6b910: Loading layer [==================>]  71.68kB/71.68kB
+Loaded image: only-hello:qn5x1pnk7d467jsl81jng7168qsks42l
+<span class="nix-command">docker run only-hello:qn5x1pnk7d467jsl81jng7168qsks42l hello"</span>
+Hello, world</pre>
+        <p>
+          Learn more <a href="[%root%]nixpkgs/manual/#sec-pkgs-dockerTools">how
+            to build docker images</a>.
+        </p>
+
+      </div>
+      <div class="span6">
+
+        <h2>Declarative cloud images</h2>
+        <p>How hard would it be to build and configure a <strong>Amazon EC2</strong> image?</p>
+        <p>
+          With the following <code>amazon.nix</code> we defined nginx which is
+          serving example <code>/var/www</code> folder, having a valid ssl
+          certificate (via LetsEncrypt) and enabled recommended security
+          settings.
+        </p>
+        <pre class="well">
+{ pkgs, ...}:
+{
+  security.acme.acceptTerms = true;
+  security.acme.email = "nix@example.com";
+  services.nginx = {
+    enable = true;
+    recommendedGzipSettings = true;
+    recommendedOptimisation = true;
+    recommendedProxySettings = true;
+    recommendedTlsSettings = true;
+    virtualHosts."example.com" = {
+      enableACME = true;
+      forceSSL = true;
+      locations."/".root = "/var/www";
```

Comment:
this still needs content

Reply:
can nginx even start when it is missing?

you could serve the nginx default page like debian does after installing nginx package

https://github.com/davidak/nixos-config/blob/02f09fafaa306aff83e61b7d36fcc1dc0fb476bf/machines/web/configuration.nix#L127-L130